lib/Search/SearchResults.php

```diff
@@ -168,13 +168,18 @@ protected function updateObjLinks(SugarBean $obj, &$fieldDef)
     {
         if ($fieldDef['type'] == 'relate' && isset($fieldDef['link']) && isset($fieldDef['id_name']) && $fieldDef['id_name']) {
             $relId = $this->getRelatedId($obj, $fieldDef['id_name'], $fieldDef['link']);
-            $obj->{$fieldDef['name']} = $this->getLink($obj->{$fieldDef['name']}, $fieldDef['module'], $relId, 'DetailView');
+            if (!empty($relId)) {
+                $obj->{$fieldDef['name']} = $this->getLink($obj->{$fieldDef['name']}, $fieldDef['module'], $relId,
+                    'DetailView');
+            } else {
+                LoggerManager::getLogger()->warn('Unresolved related ID for field: ' . $fieldDef['id_name']);
+            }
```

Comment:
@Dillon-Brown @mattlorimer 

Here I would keep the `LoggerManager::getLogger()->warn('Unresolved related ID for field: ` within the `getRelatedId` even if in 2 places. As it log even it the `getRelatedId` is called by other methods in the future

Reply:
Agreed, pushed!

Reply:
Thanks!

Reply:
@Dillon-Brown since added the logging back to the method we can remove it from here.

Reply:
Pushed

Reply:
thanks!